components/system_stats/src/lib.rs

```diff
@@ -0,0 +1,158 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Helps to collect and report statistics about the system.
+
+use std::{sync::Mutex, time::Duration};
+
+pub use sysinfo::LoadAvg;
+use sysinfo::{Cpu, CpuRefreshKind, MemoryRefreshKind, RefreshKind, System};
+
+/// The stats about the system.
+#[derive(Debug)]
+pub struct SystemStats {
+    /// The usage's range is [0, 1.0]
+    pub cpu_usage: f32,
+    /// The memory is counted in byte.
+    pub used_memory: u64,
+    /// The memory is counted in byte.
+    pub total_memory: u64,
+    pub load_avg: LoadAvg,
```

Comment:
Should we expose primitive types in sysinfo lib? Maybe defining a new type is more appropriate?

Reply:
I've re-exported the `LoadAvg` in the library. And I guess no such need for `LoadAvg` is simple enough.